File: common/src/main/java/com/kt/kol/common/exception/BusinessException.java
package com.kt.kol.common.exception;

public class BusinessException extends RuntimeException {

	private static final long serialVersionUID = -2291025218301636662L;

	private String errorCode;

	public BusinessException(String message) {
		super(message);
	}

	public BusinessException(String errorCode, String message) {
		super(message);
		this.errorCode = errorCode;
	}

	public BusinessException(String message, Throwable cause) {
		super(message, cause);
	}

	public String getErrorCode() {
		return errorCode;
	}

}

File: common/src/main/kotlin/com/kt/kol/common/exception/BusinessException.kt
package com.kt.kol.common.exception

class BusinessException : RuntimeException {

    val errorCode: String?

    constructor(message: String) : super(message) {
        this.errorCode = null
    }

    constructor(errorCode: String, message: String) : super(message) {
        this.errorCode = errorCode
    }

    constructor(message: String, cause: Throwable) : super(message, cause) {
        this.errorCode = null
    }

    constructor(errorCode: String, message: String, cause: Throwable) : super(message, cause) {
        this.errorCode = errorCode
    }

    companion object {
        private const val serialVersionUID = -2291025218301636662L
    }
}